rust/ql/test/library-tests/dataflow/sources/test.rs

```diff
@@ -298,99 +499,355 @@ fn test_io_fs() -> std::io::Result<()> {
+                Err(e) => {
+                    println!("error: {}", e);
+                    break;
+                }
+            }
+        }
+    } else {
+        let mut buffer = Vec::new();
+        loop {
+            match tokio_stream.try_read_buf(&mut buffer) {
+                Ok(0) => {
+                    println!("end");
+                    break;
+                }
+                Ok(_n) => {
+                    println!("buffer = {:?}", buffer);
+                    sink(&buffer); // $ hasTaintFlow=address
+                    break; // (or we could wait for more data)
+                }
+                Err(ref e) if e.kind() == std::io::ErrorKind::WouldBlock => {
+                    // wait...
+                    continue;
+                }
+                Err(e) => {
+                    println!("error: {}", e);
+                    break;
+                }
+            }
+        }
+    }
+
+    Ok(())
+}
+
+async fn test_std_to_tokio_tcpstream() -> std::io::Result<()> {
+    // using tokio::io to fetch a web page
+    let address = "example.com:80";
+
+    // create the connection
+    println!("connecting to {}...", address);
+    let std_stream = std::net::TcpStream::connect(address)?; // $ Alert[rust/summary/taint-sources]
+
+    // convert to tokio stream
+    std_stream.set_nonblocking(true)?;
+    let mut tokio_stream = tokio::net::TcpStream::from_std(std_stream)?;
+
+    // send request
+    tokio_stream.write_all(b"GET / HTTP/1.1\nHost:example.com\n\n").await?;
+
+    // read response
+    let mut buffer = vec![0; 32 * 1024];
+    let _n = tokio_stream.read(&mut buffer).await?; // $ MISSING: Alert[rust/summary/taint-sources]
+
+    println!("data = {:?}", buffer);
+    sink(&buffer); // $ MISSING: hasTaintFlow
+    sink(buffer[0]); // $ MISSING: hasTaintFlow
+
+    Ok(())
+}
+
 #[tokio::main]
 async fn main() -> Result<(), Box<dyn std::error::Error>> {
     let case = std::env::args().nth(1).unwrap_or(String::from("1")).parse::<i64>().unwrap(); // $ Alert[rust/summary/taint-sources]
 
+    println!("test_env_vars...");
+    test_env_vars();
+
+    /*println!("test_env_args...");
+    test_env_args();*/
+
+    println!("test_env_dirs...");
+    test_env_dirs();
+
+    /*println!("test_reqwest...");
+    match futures::executor::block_on(test_reqwest()) {
+        Ok(_) => println!("complete"),
+        Err(e) => println!("error: {}", e),
+    }*/
+
     println!("test_hyper_http...");
     match futures::executor::block_on(test_hyper_http(case)) {
         Ok(_) => println!("complete"),
         Err(e) => println!("error: {}", e),
     }
-    println!("");
+
+    /*println!("test_io_stdin...");
+    match test_io_stdin() {
+        Ok(_) => println!("complete"),
+        Err(e) => println!("error: {}", e),
+    }
+
+    println!("test_tokio_stdin...");
+    match futures::executor::block_on(test_tokio_stdin()) {
+        Ok(_) => println!("complete"),
+        Err(e) => println!("error: {}", e),
+    }*/
```

Comment:
Why is this commented out?

Reply:
I believe just because I was running the test to check some of the web-based stuff really works, and it was a pain having to type stuff into `stdin` to get to it.  Happy to uncomment it now.

Reply:
Updated.